kafka/error.go

```diff
@@ -40,7 +40,8 @@ func newError(code C.rd_kafka_resp_err_t) (err Error) {
 	return Error{ErrorCode(code), "", false}
 }
 
-func newGoError(code ErrorCode) (err Error) {
+// NewError creates a new Error with specified ErrorCode.
+func NewError(code ErrorCode) (err Error) {
```

Comment:
Huhm, I wonder if you should expose fatal as well, it makes it possible to test fatal error handling which is important for idempotent producer apps.
Can you add a fatal argument?

Reply:
Nice idea, will fix